libraries/AP_Scripting/applets/Aerobatics/FixedWing/RateBased/sport_aerobatics.lua

```diff
@@ -460,7 +460,7 @@ end
 local knife_edge_ms = 0
 function do_knife_edge(arg1,arg2)
   -- arg1 is angle +/-180, duration is arg2
-    local now = millis():tofloat() * 0.001
+    local now = seconds_since_boot()
```

Comment:
Why not just correct this to do the time wrapping correctly with the uint32 case? It's a simple replacement, and more accurate.

Reply:
using uint32 is both slow and less accurate. The seconds_since_boot() is microsecond accurate, and cheaper

Reply:
Don't you lose all the accuracy immediately when you don't have double precision support available though?

Reply:
23 bit mantissa does not leave a lot of space on single precision float, I'd actually expect this to be significantly worse then before in accuracy on any 32bit board....

Reply:
As discussed on the dev call - isn't it really time to recognize that it's not really worthwhile supporting scripting on these low end boards?

Reply:
> the point is to get 64 bit float on any board running a non-trivial script

I kind of get it, but at the same time I think there's a lot of value on say more lightly loaded Periph's (on M4's) to not forcing everything down this path. And it seems odd to chase the precision here when the core controls aren't at this resolution.

I do support a time wrapper utility if needed, but I'd almost prefer to see that just be us provide a library with get delta functions and the like. My big objection is I just don't want to force M4's to use double precision in scripting, as it hamstrings some of the AP_Periph style uses, and if we encourage the default path to be "rely upon high precision" that makes things harder to do. (I think at that point we also need a binding of "does this board have double support")